src/VisualStudio/Core/Def/Implementation/Watson/WatsonExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Telemetry;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.ErrorReporting
+{
+    internal static class WatsonExtensions
+    {
+        /// <summary>
+        /// This sets extra watson bucket parameters to make bucketting better
+        /// in non fatal watson report
+        /// </summary>
+        public static void SetExtraParameters(this IFaultUtility fault, Exception exception)
+        {
+            switch (exception)
+            {
+                case RemoteInvocationException remote:
+                    fault.SetBucketParameter(8, remote.GetParameterString());
+                    return;
+                case AggregateException aggregate:
+                    if (aggregate.InnerException == null)
+                    {
+                        return;
+                    }
+                    else if (aggregate.InnerExceptions.Count == 1)
+                    {
+                        fault.SetBucketParameter(8, aggregate.GetParameterString());
```

Comment:
>8 [](start = 49, length = 1)

What are those "7" and "8"? Can we introduce constants for them and make comments explaining their meanings?

Reply:
7,8 bucket slot for NFW Watson is reserved slots for customization like what I did here. when not customized, it is just empty slot. 7 or 8 doesn't have some semantic meaning attached to it.

Reply:
Thank you! Those comments will also be very helpful in code. Could you please add them there?

Reply:
sure will do.